mnefun/_mnefun.py

```diff
@@ -1812,20 +1869,19 @@ def gen_inverses(p, subjects, run_indices):
             out_flags += ['-meg-eeg']
             meg_bools += [True]
             eeg_bools += [True]
-        # maybe we should only do this if p.cov_rank == 'full'?
-        rank = _compute_rank(p, subj, run_indices[si])
+        if p.cov_rank == 'full' and p.compute_rank:
+            rank = _compute_rank(p, subj, run_indices[si])
+        else:
+            rank = None  # should be safe from our gen_covariances step
         if make_erm_inv:
+            # We now process the empty room with "movement
+            # compensation" so it should get the same rank!
             erm_name = op.join(cov_dir, safe_inserter(p.runs_empty[0], subj) +
                                p.pca_extra + p.inv_tag + '-cov.fif')
             empty_cov = read_cov(erm_name)
-            if empty_cov.get('method', 'empirical') == 'empirical' or \
-                    p.cov_rank != 'full':
-                kwargs = dict()
-                # The empty-room covariance can have a different spatial
-                # pattern
-                if 'rank' in get_args(regularize):
-                    kwargs['rank'] = 'full'
-                empty_cov = regularize(empty_cov, raw.info, **kwargs)
+            if p.force_erm_cov_rank_full and p.cov_method == 'empirical':
+                empty_cov = regularize(
+                    empty_cov, epochs.info, rank='full')
```

Comment:
Does it make sense to put in an assertion here to make sure covs have same rank?

Reply:
Actually this step forces it to be full rank, in case you want it to be that way